### PR TITLE
store serial numbers in device descriptor (not in  global GPS data)

### PR DESCRIPTION
**Pull request by AlphaLima**
Important if two devices are connected, otherwise the DoStatusMessage is toggeling between the devices

fixed vario source in status dialog, same problem

### CODE REVIEW

File: Common/Source/Calc/Vario.cpp [hunk 1]
```diff
@@ -19,7 +19,8 @@ bool VarioAvailable(const NMEA_INFO& Info) {
 }
 
 void UpdateVarioSource( NMEA_INFO& Info, const DeviceDescriptor_t& d, double Vario) {
-  if((unsigned)d.PortNumber <= Info.VarioSourceIdx) {
+  if((unsigned)d.PortNumber <= Info.VarioSourceIdx) 
+  {
```
Comment (brunotl) [inline Common/Source/Calc/Vario.cpp:23]: why ?

File: Common/Source/Dialogs/dlgStatus.cpp [hunk 2]
```diff
@@ -239,22 +239,17 @@ static void UpdateValuesSystem() {
     if (VarioAvailable(GPS_INFO)) {
 	// LKTOKEN  _@M199_ = "Connected"
 #ifdef DEVICE_SERIAL
-     if(GPS_INFO.HardwareId >0)
+     if(DeviceList[ GPS_INFO.VarioSourceIdx ].HardwareId >0)
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgStatus.cpp:242]: GPS_INFO and DeviceList is shared data, mutex lock is required.

File: Common/Header/device.h [hunk 3]
```diff
@@ -167,7 +167,11 @@ struct DeviceDescriptor_t {
   unsigned ErrTx;
   // Com ports hearth beats, based on LKHearthBeats
   unsigned HB;
-  
+#ifdef DEVICE_SERIAL
+  int HardwareId;
+  int SerialNumber;
+  double SoftwareVer;
+#endif
```
Comment (brunotl) [inline Common/Header/device.h:174]: why store number instead of original string ?

File: Common/Source/Calc/Vario.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/Calc/Vario.cpp:23]: forgot to restore! can stay original

File: Common/Source/Dialogs/dlgStatus.cpp (hunk 2 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgStatus.cpp:242]: How, correctly?

File: Common/Header/device.h (hunk 3 above)
Comment (AlphaLima) [reply Common/Header/device.h:174]: Just moved it from the without changing data format.

**Comment by brunotl** [comment]
don't build if "DEVICE_SERIAL" is not defined

**Pull request closed without merging**